### DISCUSSION TITLE
No image in production build

### DISCUSSION THREAD

**Discussion by Defite** [gatsbyjs/gatsby#34553] [Help]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

Updated Gatsby.js from v2 to v4 and know all frontmatter images that worked before disappeared from production build, but are present in dev mode.

It's worth mentioning that I use `gatsby-remark-relative-source` in order to store all images for posts in separate folder. My content is in `content/articles/hello-world.mdx` and my images are in `/uploads` folder.

### Reproduction Link

https://gatsby-webgrower.vercel.app/articles/hello-world

### Steps to Reproduce

1. Open https://gatsby-webgrower.vercel.app/articles/hello-world
2. There must be an photo of my Volvo, but it's not there
...

### Expected Result

Photo is in text

### Actual Result

There is no photo

### Environment

```shell
System:
    OS: macOS 12.0.1
    CPU: (8) x64 Intel(R) Core(TM) i7-4770HQ CPU @ 2.20GHz
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 12.22.1 - ~/.nvm/versions/node/v12.22.1/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 7.21.1 - ~/.nvm/versions/node/v12.22.1/bin/npm
  Languages:
    Python: 2.7.18 - /usr/bin/python
  Browsers:
    Chrome: 97.0.4692.71
    Firefox: 94.0.1
    Safari: 15.1
  npmPackages:
    gatsby: ^4.5.3 => 4.5.3 
    gatsby-image: ^3.11.0 => 3.11.0 
    gatsby-plugin-manifest: ^4.5.2 => 4.5.2 
    gatsby-plugin-offline: ^5.5.2 => 5.5.2 
    gatsby-plugin-postcss: ^5.5.0 => 5.5.0 
    gatsby-plugin-react-helmet: ^5.5.0 => 5.5.0 
    gatsby-plugin-react-svg: ^3.1.0 => 3.1.0 
    gatsby-plugin-sharp: ^4.5.2 => 4.5.2 
    gatsby-remark-images: ^6.5.2 => 6.5.2 
    gatsby-remark-relative-source: ^0.0.5 => 0.0.5 
    gatsby-remark-responsive-iframe: ^5.5.0 => 5.5.0 
    gatsby-source-filesystem: ^4.5.2 => 4.5.2 
    gatsby-transformer-remark: ^5.5.2 => 5.5.2 
    gatsby-transformer-sharp: ^4.5.0 => 4.5.0
```

**Comment by LekoArts** [comment]
Hi!

This is most likely an issue with `gatsby-remark-relative-source` - I saw that the owner of the package doesn't allow issues on its repo, so you can't really ask for feedback there. If you can give a minimal reproduction of your setup we might be able to give you insturctions on how to do things without this plugin and have it working.

**Reply by Defite** [reply]
I've already attached reproduction link. I thought about `gatsby-remark-relative-source` too. So, is there any way to keep my content in one folder and images in another? What should I do for that?

gatsby-config.js
```js
{
     resolve: 'gatsby-source-filesystem',
     options: {
	name: 'uploads',
	path: `${__dirname}/uploads`,
     },
    },
    {
      resolve: 'gatsby-source-filesystem',
      options: {
        name: 'pages',
        path: `${__dirname}/content/pages`,
      },
    },
    {
      resolve: 'gatsby-source-filesystem',
      options: {
        name: 'articles',
        path: `${__dirname}/content/articles`,
      },
    },
    {
      resolve: 'gatsby-source-filesystem',
      options: {
        name: 'links',
        path: `${__dirname}/content/links`,
      },
    },
```

Example md file:

```md
---
title: Привет, мир!
path: "/articles/hello-world"
headerImg: "/uploads/6c7045ds-1920.jpg" <-- this image disappeared from prod build
---
```

**Reply by LekoArts** [reply]
By reproduction I mean a GitHub repo with a project that one could run :)

If you're only using images in the frontmatter, then that's all what the plugin is doing:

https://github.com/SublimateDesign/gatsby-remark-relative-source/blob/master/src/index.js#L152-L175

You could remove the plugin, copy over the function to your `gatsby-node` file where you're using it and debug it. Or you'll use schema customization as explained here: https://byderek.com/post/a-stackoverflow-question--a-use-case-for-gatsbys-field-extension

**Reply by Defite** [reply]
Ah, repo, of course. https://github.com/webgrower/gatsby

And I'll try debugging, of course, thanks for advice.

**Reply by Defite** [reply]
Strange thing. I tried to do this:

```js
exports.createSchemaCustomization = ({ actions }) => {
  const { createTypes } = actions

  const typeDefs = `
		type MarkdownRemark implements Node {
			frontmatter: Frontmatter
		}
		type Frontmatter @infer {
      linkUrl: String
      headerImg: File @fileByAbsolutePath(path: "uploads/")
		}
	`

  createTypes(typeDefs)
}
```

and got an error:

```bash
Field extension `fileByAbsolutePath` on `Frontmatter.headerImg` is not available.
```

Why?

**Reply by Defite** [reply]
> Ah, repo, of course. https://github.com/webgrower/gatsby
> 
> And I'll try debugging, of course, thanks for advice.

Tried debugging, then tried solution from article - nothing works. I got `null` from `headerImg`.

**Comment by Defite** [comment]
Looks like no one is interested in helping me, so I've moved to Next.js. Thanks.
